core/lib/libtomcrypt/sm2-pke.c

```diff
@@ -255,14 +255,16 @@ TEE_Result crypto_acipher_sm2_pke_decrypt(struct ecc_keypair *key,
 		goto out;
 	}
 
-	if (mp_unsigned_bin_size(x2y2p->x) != 32 ||
-	    mp_unsigned_bin_size(x2y2p->y) != 32) {
+	if (mp_unsigned_bin_size(x2y2p->x) > 32 ||
+	    mp_unsigned_bin_size(x2y2p->y) > 32) {
 		res = TEE_ERROR_BAD_STATE;
 		goto out;
 	}
 
-	mp_to_unsigned_bin(x2y2p->x, x2y2);
-	mp_to_unsigned_bin(x2y2p->y, x2y2 + 32);
+	mp_to_unsigned_bin(x2y2p->x, x2y2 +
```

Comment:
Please comment on the layout of `x2y2`. Why is the first `+ 32` needed for instance.

Reply:
The first `+ 32` is not on its own, it is `32 - mp_unsigned_bin_size(x2y2p->x)`. So in the end `x2y2` is just the coordinates `(x2, y2)` of point `x2y2p` formatted as two consecutive big endian integers. It should be obvious if I use my proposed `mp_to_unsigned_bin2()`.